### PR TITLE
make assemblystream seekable

### LINKED ISSUE #14618: Make assembly stream seekable
I guess it would make sense to make the assembly stream seekable. that would allow al implementations to just use the stream as is.

CC: @icewind1991

### PR DESCRIPTION
**Pull request by icewind1991**
Fixes #14618

### CODE REVIEW

File: apps/dav/lib/Upload/AssemblyStream.php [hunk 1]
```diff
@@ -83,12 +83,47 @@ public function stream_open($path, $mode, $options, &$opened_path) {
...
-		return false;
+		if ($whence === SEEK_CUR) {
+			$offset = $this->stream_tell() + $offset;
+		} else if ($whence === SEEK_END) {
+			$offset = $this->size + $offset;
+		}
+
+		if ($offset > $this->size) {
+			throw new \Exception('invalid offset');
+			return false;
```
Comment (MorrisJobke) [inline apps/dav/lib/Upload/AssemblyStream.php]: 🧐

Comment (icewind1991) [reply apps/dav/lib/Upload/AssemblyStream.php]: :see_no_evil:

**Review by rullzer — APPROVED** [review]
Code looks good.
I would welcome more tests

**Review by MorrisJobke — APPROVED** [review]
Code looks good 👍

**Pull request merged**